### PR TITLE
Add multitask and few-shot learning sections

### PR DESCRIPTION
**Pull request by jaclyn-taroni**
### Purpose

Addresses #83.

### Directions for reviewers

I am not that happy about where this is at the moment. Part of the struggle, for me, is the breadth of multitask and few-shot learning that are beyond the scope of what could possibly fit here. It would be helpful for reviewers to comment on what they would like to see expanded or contracted.  The main point I am trying to get across here, as with the transfer learning question, is that it depends™️   when it comes to usefulness and performance. 

### Manuscript checklist

Unless otherwise noted above, this PR will be considered ready for review when all four items have been checked.

- [x] All changes to text follow "one sentence per line" [[Manubot instructions](https://github.com/jaybee84/ml-in-rd/blob/draft-branch/USAGE.md#manubot-markdown)]
- [x] All citations follow the [Manubot citation instructions](https://github.com/jaybee84/ml-in-rd/blob/draft-branch/USAGE.md#citations)
- [x] All changes or additions to tables follow the [Manubot table instructions](https://github.com/jaybee84/ml-in-rd/blob/draft-branch/USAGE.md#tables)
- [x] All figures follow the [Manubot figure instructions](https://github.com/jaybee84/ml-in-rd/blob/draft-branch/USAGE.md#figures)
- [ ] All terms that need to be added to glossary have been added to #50 
- [ ] There are no new spelling errors identified by the Manubot spellchecker

### CODE REVIEW

File: content/05.prior-knowledge.md [hunk 1]
```diff
@@ -90,4 +92,30 @@ MultiPLIER has additional attributes that make it practical for studying rare di
...
-In the authors' opinion, selecting an appropriate model for a given task and evaluations that are well-aligned with a research goal are crucial for applying these approaches in rare diseases.
\ No newline at end of file
+In the authors' opinion, selecting an appropriate model for a given task and evaluations that are well-aligned with a research goal are crucial for applying these approaches in rare diseases.
+
+##### Multitask and few-shot learning
+ 
+Where transfer learning can be supervised or unsupervised, the related approaches multitask and few-shot learning are forms of supervised learning that generally rely on deep neural networks.
+Multitask learning is an approach where classifiers are learned for _related tasks_ at the same time using a shared representation [@doi:10.1023/A:1007379606734].
+Few-shot learning is the generalization of a model trained on related data and tasks to a new task with limited labeled data (e.g., the detection of a patient with a rare disease from a low number of examples of that rare disease).
```
Comment (jaybee84) [inline content/05.prior-knowledge.md]:
```suggestion
Few-shot learning is the generalization of a model trained on related tasks to a new task with limited labeled data (e.g., the detection of a patient with a rare disease from a low number of examples of that rare disease). 
```

File: content/05.prior-knowledge.md [hunk 2]
```diff
@@ -90,4 +92,30 @@ MultiPLIER has additional attributes that make it practical for studying rare di
...
-In the authors' opinion, selecting an appropriate model for a given task and evaluations that are well-aligned with a research goal are crucial for applying these approaches in rare diseases.
\ No newline at end of file
+In the authors' opinion, selecting an appropriate model for a given task and evaluations that are well-aligned with a research goal are crucial for applying these approaches in rare diseases.
+
+##### Multitask and few-shot learning
+ 
+Where transfer learning can be supervised or unsupervised, the related approaches multitask and few-shot learning are forms of supervised learning that generally rely on deep neural networks.
+Multitask learning is an approach where classifiers are learned for _related tasks_ at the same time using a shared representation [@doi:10.1023/A:1007379606734].
+Few-shot learning is the generalization of a model trained on related data and tasks to a new task with limited labeled data (e.g., the detection of a patient with a rare disease from a low number of examples of that rare disease). 
+As with transfer learning, multitask and few-shot learning can be applied in situations where there are insufficient data for a given condition.
```
Comment (jaybee84) [inline content/05.prior-knowledge.md]:
```suggestion

```

File: content/05.prior-knowledge.md [hunk 3]
```diff
@@ -90,4 +92,30 @@ MultiPLIER has additional attributes that make it practical for studying rare di
...
+Like rare disease research, drug discovery suffers from a relatively small amount of data for a given task.
+We highlight a few applications of multitask or few-shot learning in the drug discovery space, in addition to rare disease or condition research, to illustrate key takeaways.
+
+Multitask neural networks that predict multiple tasks simultaneously are generally thought to improve performance over models that make predictions for a single task by learning a shared representation and effectively being exposed to more training data than the single task case [@doi:10.1023/A:1007379606734; @arxiv:1606.08793].
+Kearnes, Goldman, and Pande set out to examine the effects of dataset size and task relatedness on multitask learning performance improvements ("multitask effect") in an industrial drug discovery setting [@arxiv:1606.08793].
```
Comment (jaybee84) [inline content/05.prior-knowledge.md]:
```suggestion
Kearnes, Goldman, and Pande set out to examine the effects of dataset size and task relatedness on multitask learning performance improvements ("multitask effect") in drug discovery (where data is also sparse similar to rare disease) [@arxiv:1606.08793].
```

File: content/05.prior-knowledge.md [hunk 4]
```diff
@@ -90,4 +92,30 @@ MultiPLIER has additional attributes that make it practical for studying rare di
...
+
+##### Multitask and few-shot learning
+ 
+Where transfer learning can be supervised or unsupervised, the related approaches multitask and few-shot learning are forms of supervised learning that generally rely on deep neural networks.
+Multitask learning is an approach where classifiers are learned for _related tasks_ at the same time using a shared representation [@doi:10.1023/A:1007379606734].
+Few-shot learning is the generalization of a model trained on related data and tasks to a new task with limited labeled data (e.g., the detection of a patient with a rare disease from a low number of examples of that rare disease). 
+As with transfer learning, multitask and few-shot learning can be applied in situations where there are insufficient data for a given condition.
+Like rare disease research, drug discovery suffers from a relatively small amount of data for a given task.
+We highlight a few applications of multitask or few-shot learning in the drug discovery space, in addition to rare disease or condition research, to illustrate key takeaways.
```
Comment (jaybee84) [inline content/05.prior-knowledge.md]:
```suggestion

```

File: content/05.prior-knowledge.md [hunk 5]
```diff
@@ -90,4 +92,30 @@ MultiPLIER has additional attributes that make it practical for studying rare di
...
\ No newline at end of file
+In the authors' opinion, selecting an appropriate model for a given task and evaluations that are well-aligned with a research goal are crucial for applying these approaches in rare diseases.
+
+##### Multitask and few-shot learning
+ 
+Where transfer learning can be supervised or unsupervised, the related approaches multitask and few-shot learning are forms of supervised learning that generally rely on deep neural networks.
+Multitask learning is an approach where classifiers are learned for _related tasks_ at the same time using a shared representation [@doi:10.1023/A:1007379606734].
+Few-shot learning is the generalization of a model trained on related data and tasks to a new task with limited labeled data (e.g., the detection of a patient with a rare disease from a low number of examples of that rare disease). 
+As with transfer learning, multitask and few-shot learning can be applied in situations where there are insufficient data for a given condition.
+Like rare disease research, drug discovery suffers from a relatively small amount of data for a given task.
```
Comment (jaybee84) [inline content/05.prior-knowledge.md]:
```suggestion

```

File: content/05.prior-knowledge.md [hunk 6]
```diff
@@ -90,4 +92,30 @@ MultiPLIER has additional attributes that make it practical for studying rare di
...
+The authors found that the multitask performance gains were highly dataset-specific: smaller datasets tended to benefit most from multitask learning and the addition of more training data did not guarantee improved performance for multitask models.
+In predicting phenotypes from EHR data, Ding et al. demonstrated that multitask neural networks outperformed single-task networks for predicting complex rare phenotypes but not common phenotypes [@arxiv:1808.03331].
+Liu et al. developed a method to train long short-term memory networks, a type of recurrent neural network, to predict mortality in rare diseases using EHR data as input [@arxiv:2004.05318v2].
+Their method, Ada-SiT (Adaptation to Similar Tasks), was specifically designed for many tasks with insufficient data and allowed for task similarity to be measured during training.
+
+Few-shot learning, also referred to as one-shot learning, relies on using prior knowledge to generalize to new prediction tasks where there are a low number of examples [@arxiv:1904.05046v3], where a distance metric is learned from input data and used to compare new examples for prediction [@doi:10.1021/acscentsci.6b00367].
```
Comment (jaybee84) [inline content/05.prior-knowledge.md]:
```suggestion
On the other hand, Few-shot learning or one-shot learning relies on using prior knowledge to generalize to new prediction tasks where there are a low number of examples [@arxiv:1904.05046v3], where a distance metric is learned from input data and used to compare new examples for prediction [@doi:10.1021/acscentsci.6b00367].
```

File: content/05.prior-knowledge.md [hunk 7]
```diff
@@ -90,4 +92,30 @@ MultiPLIER has additional attributes that make it practical for studying rare di
...
+Altae-Tran et al. developed a method for predicting small molecule activity that learned a meaningful distance over compounds [@doi:10.1021/acscentsci.6b00367].
```
Comment (jaybee84) [inline content/05.prior-knowledge.md]:
```suggestion
Altae-Tran et al. developed a method for predicting small molecule activity that learned a meaningful distance metric over the properties of various compounds [@doi:10.1021/acscentsci.6b00367].
```

File: content/05.prior-knowledge.md [hunk 8]
```diff
@@ -90,4 +92,30 @@ MultiPLIER has additional attributes that make it practical for studying rare di
...
+However, the authors results suggested underperformance of one-shot learning methods relative to baseline random forest models when structural similarity could not be exploited and did not show support for generalization of models trained on very different contexts from the target task.
```
Comment (jaybee84) [inline content/05.prior-knowledge.md]:
```suggestion
However, the authors' results suggested underperformance of one-shot learning methods relative to baseline random forest models when structural similarity could not be exploited and did not show support for generalization of models trained on very different contexts from the target task.
```

File: content/05.prior-knowledge.md [hunk 9]
```diff
@@ -90,4 +92,30 @@ MultiPLIER has additional attributes that make it practical for studying rare di
...
+Quellec et al. presented a few-shot learning approach for detecting rare pathologies in fundus photographs [@arxiv:1907.09449v3].
+The authors train a convolutional neural network (CNN) to predict common pathologies, which tends to cluster similar conditions in feature space; the learned feature space is used to train a probabilistic model for each rare pathology.
```
Comment (jaybee84) [inline content/05.prior-knowledge.md]:
```suggestion
The authors trained a convolutional neural network (CNN) to predict common pathologies, which tended to cluster similar conditions in feature space. 
The learned feature space was then used to train a probabilistic model for each rare pathology.
```

File: content/05.prior-knowledge.md [hunk 10]
```diff
@@ -90,4 +92,30 @@ MultiPLIER has additional attributes that make it practical for studying rare di
...
+This approach outperformed multitask learning, which suggests few-shot learning provides some advantages when training to predict common conditions simultaneously results in a loss of performance [@arxiv:1907.09449v3].
```
Comment (jaybee84) [inline content/05.prior-knowledge.md]:
```suggestion
This approach outperformed multitask learning, which suggests few-shot learning provides some advantages when training to predict common conditions, but simultaneously results in a loss of performance [@arxiv:1907.09449v3].  
```

File: content/05.prior-knowledge.md [hunk 11]
```diff
@@ -90,4 +92,30 @@ MultiPLIER has additional attributes that make it practical for studying rare di
...
 In the natural images field, researchers have demonstrated that the transferability of features depends on relatedness of tasks [@arxiv:1411.1792].
 The limits of transfer learning for and the concept of relatedness in high-dimensional biomedical data assaying rare diseases are open research questions.
-In the authors' opinion, selecting an appropriate model for a given task and evaluations that are well-aligned with a research goal are crucial for applying these approaches in rare diseases.
\ No newline at end of file
+In the authors' opinion, selecting an appropriate model for a given task and evaluations that are well-aligned with a research goal are crucial for applying these approaches in rare diseases.
+
+##### Multitask and few-shot learning
+ 
+Where transfer learning can be supervised or unsupervised, the related approaches multitask and few-shot learning are forms of supervised learning that generally rely on deep neural networks.
+Multitask learning is an approach where classifiers are learned for _related tasks_ at the same time using a shared representation [@doi:10.1023/A:1007379606734].
```
Comment (jaybee84) [inline content/05.prior-knowledge.md]: A sentence here to describe what is meant by a "task" would be very helpful here.

File: content/05.prior-knowledge.md [hunk 12]
```diff
@@ -90,4 +92,30 @@ MultiPLIER has additional attributes that make it practical for studying rare di
...
+Altae-Tran et al. developed a method for predicting small molecule activity that learned a meaningful distance over compounds [@doi:10.1021/acscentsci.6b00367].
+However, the authors results suggested underperformance of one-shot learning methods relative to baseline random forest models when structural similarity could not be exploited and did not show support for generalization of models trained on very different contexts from the target task.
+Quellec et al. presented a few-shot learning approach for detecting rare pathologies in fundus photographs [@arxiv:1907.09449v3].
+The authors train a convolutional neural network (CNN) to predict common pathologies, which tends to cluster similar conditions in feature space; the learned feature space is used to train a probabilistic model for each rare pathology.
+This approach outperformed multitask learning, which suggests few-shot learning provides some advantages when training to predict common conditions simultaneously results in a loss of performance [@arxiv:1907.09449v3].  
+
+Multitask and few-shot learning are comprised of a variety of approaches and architectures that are beyond this scope of this work.
```
Comment (jaybee84) [inline content/05.prior-knowledge.md]: adding a few names and refs of the approaches/architectures here without going into details can help the audience find and explore the details if they want.

**Review by jaybee84 — APPROVED** [review]
Added some minor comments, otherwise looks good to me.

File: content/05.prior-knowledge.md (hunk 10 above)
Comment (jaclyn-taroni) [reply content/05.prior-knowledge.md]: Reworded this sentence in https://github.com/jaybee84/ml-in-rd/pull/99/commits/522d8a85e0170b4cefeeb3d07ddd41aea83a177d.

File: content/05.prior-knowledge.md (hunk 11 above)
Comment (jaclyn-taroni) [reply content/05.prior-knowledge.md]: I tacked something on to the end of this sentence in https://github.com/jaybee84/ml-in-rd/pull/99/commits/a2bd1a613626cd9ff0afdce6609b5bca3ee7e3e2.

File: content/05.prior-knowledge.md (hunk 12 above)
Comment (jaclyn-taroni) [reply content/05.prior-knowledge.md]: Opted to add a few citations for survey articles in https://github.com/jaybee84/ml-in-rd/pull/99/commits/33c9f2c3bfca8a26d4aab5ece5abe2febbabc894, but am happy to revisit during one of the rounds of editing!

**Comment by jaclyn-taroni** [comment]
Thanks for the feedback @jaybee84! I've incorporated changes in response to your comments. Once the checks pass, I am planning to merge this so we can continue our edits 🔁 !

**Pull request merged**